src/Microsoft.Azure.Batch.SoftwareEntitlement.Common/X509Certificate2Extensions.cs

```diff
@@ -8,6 +10,8 @@ namespace Microsoft.Azure.Batch.SoftwareEntitlement.Common
     /// </summary>
     public static class X509Certificate2Extensions
     {
+        private const string ServerAuthenticationOid = "1.3.6.1.5.5.7.3.1";
```

Comment:
Where does this ID come from? Would be useful to have a comment with a URL to some docs.

Reply:
I failed to find anything very authoritative at the time. Now I have ( https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.server.kestrel.https.httpsconnectionadapteroptions?view=aspnetcore-2.2#properties ) it turns out I haven't been checking the condition correctly.

The EKU only needs to _contain_ (not be) this OID, and only _if_ there actually is an EKU extension.

Hold up while I make some more changes...